src/Microsoft.FeatureManagement/IFeaturedService.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.FeatureManagement
+{
+    public interface IFeaturedService<TService> where TService : class
```

Comment:
Can we add comments for each input/output parameter? I am wondering the reasoning for the choice between FeaturedService vs. FeatureService.

Reply:
I think the name wasn't discussed enough actually. Thinking on the overall design that we landed on I would suggest we call this `IVariantServiceProvider` with a single method GetService.

This is similar to built-in [IServiceProvider](https://learn.microsoft.com/en-us/dotnet/api/system.iserviceprovider?view=net-8.0).

Reply:
Sounds good. I like the name `IVariantServiceProvider`

Reply:
@jimmyca15 Should the GetService method be async? Since our implementation need to get variant from feature manager through GetVariantAsync.

Reply:
@zhiyuanliang-ms yes it needs to be async. ValueTask.